egui_table/src/table.rs

```diff
@@ -181,9 +183,22 @@ pub trait TableDelegate {
     ///
     /// The [`CellInfo::row_nr`] is ignoring header rows.
     fn cell_ui(&mut self, ui: &mut Ui, cell: &CellInfo);
+
+    /// Compute the offset for the top of the given row.
+    ///
+    /// Implementing this provides a way to have arbitrary row heights. If [`Table::default_row_height`]
+    /// is not used, this method must never return `None`.
+    ///
+    /// Important:
+    /// - Must return always `None`, or always `Some(offset)`. Returning both will lead to
+    ///   inconsistent results.
+    /// - Must always return either `Some(0.0)` or `None` for `row_nr = 0`.
+    fn row_top_offset(&self, _ctx: &Context, _row_nr: u64) -> Option<f32> {
```

Comment:
If we include a `table_id: Id` parameter then row expansion state can be stored in `ctx.data`

Reply:
good call